drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java

```diff
@@ -726,29 +722,29 @@ public void setOnTemplateValueChangeCommand(Command onTemplateValueChangeCommand
     }
 
     void initDropDownData() {
-        //Set applicable flags and reference data depending upon type
-        if (DataType.TYPE_BOOLEAN.equals(this.fieldType)) {
-            this.isDropDownDataEnum = false;
-            this.dropDownData = DropDownData.create(new String[]{"true", "false"});
-        } else {
-            this.isDropDownDataEnum = true;
 
-            final Map<String, String> currentValueMap = new HashMap<String, String>();
+        this.isDropDownDataEnum = true;
 
-            if (constraintList != null && constraintList.getConstraints() != null) {
-                for (FieldConstraint con : constraintList.getConstraints()) {
-                    if (con instanceof SingleFieldConstraint) {
-                        SingleFieldConstraint sfc = (SingleFieldConstraint) con;
-                        String fieldName = sfc.getFieldName();
-                        currentValueMap.put(fieldName,
-                                            sfc.getValue());
-                    }
+        final Map<String, String> currentValueMap = new HashMap<String, String>();
+
+        if (constraintList != null && constraintList.getConstraints() != null) {
+            for (FieldConstraint con : constraintList.getConstraints()) {
+                if (con instanceof SingleFieldConstraint) {
+                    SingleFieldConstraint sfc = (SingleFieldConstraint) con;
+                    String fieldName = sfc.getFieldName();
+                    currentValueMap.put(fieldName,
+                                        sfc.getValue());
                 }
             }
+        }
 
-            this.dropDownData = oracle.getEnums(this.factType,
-                                                fieldName,
-                                                currentValueMap);
+        this.dropDownData = oracle.getEnums(this.factType,
+                                            fieldName,
+                                            currentValueMap);
+
+        if (DataType.TYPE_BOOLEAN.equals(this.fieldType) && this.dropDownData != null) {
```

Comment:
Is this `if` block correct? To me it seems we override declared enums exactly for case, customer is complaining about in ticket.

Reply:
Nope it is not. It should be `this.dropDownData == null`. I'll get it fixed.

Reply:
Fixed. Well spotted. I actually fixed this when I was about to write a test for it, but had to give up. To get a test workin I might have to edit this part more than I wanted and break it again.
The code affects enumerations for booleans in the Guided Rule Editor.

Reply:
If we decide this needs tests or changes, I would rather revert changes in this file and make a separate ticket for GRE that we fix later.